rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend

```diff
@@ -93,37 +114,47 @@ class TestStepSelector {
 
 	@Inject
 	@Optional
-	def void updateModelAndView(@EventTopic(SELECTOR_UPDATE_MODEL_AND_VIEW) Object data) {
-		logger.debug("updateModelAndView")
-		val amlInjector = amlInjectorProvider.get
-		models = amlInjector.getInstance(AmlModelsProvider).amlModels
-		Display.getDefault.syncExec[internalUpdateView]
+	def void updateView(@EventTopic(SELECTOR_UPDATE_VIEW) Object data) {
+		logger.debug("updateView for " + data)
+		if (data instanceof XtextEditor) {
+			Display.^default.syncExec[updateViewForXtextEditor(data)]
+		}
 	}
 
-	@Inject
-	@Optional
-	def void updateView(@EventTopic(SELECTOR_UPDATE_VIEW) Object data) {
-		logger.debug("updateView")
-		// TODO - check on the project of the editor
-		if (viewer.input === null) {
-			Display.^default.syncExec[internalUpdateView]
+	private def updateViewForXtextEditor(XtextEditor editor) {
+		val projectName = editor.resource.project.name
+
+		var previouslyExpandedElements = viewer.expandedElements.map[toStringPath].toSet
```

Comment:
Why is it set in line 132? It appears that the method is doing more than one thing. It's not only updating the view but also storing the previous selection in the `expandedElementsPerProject` map - but only if the project changes. Why extract this logic of updating the map into a separate method? The update might be done even if the project doesn't change.